src/tools/rbd/action/Migration.cc

```diff
@@ -62,10 +62,92 @@ static int do_abort(librados::IoCtx& io_ctx, const std::string &image_name,
 }
 
 static int do_commit(librados::IoCtx& io_ctx, const std::string &image_name,
-                     bool no_progress) {
+                     bool force, bool no_progress) {
+  librbd::image_migration_status_t migration_status;
+  int r = librbd::RBD().migration_status(io_ctx, image_name.c_str(),
+                                         &migration_status,
+                                         sizeof(migration_status));
+  if (r < 0) {
+    std::cerr << "rbd: getting migration status failed: " << cpp_strerror(r)
+              << std::endl;
+    return r;
+  }
+
+  librados::IoCtx dst_io_ctx;
+  r = librados::Rados(io_ctx).ioctx_create2(migration_status.dest_pool_id, dst_io_ctx);
+  if (r < 0) {
+    std::cerr << "rbd: accessing source pool id="
+              << migration_status.dest_pool_id << " failed: "
+              << cpp_strerror(r) << std::endl;
+    return r;
+  }
+
+  r = utils::set_namespace(migration_status.dest_pool_namespace, &dst_io_ctx);
+  if (r < 0) {
+    return r;
+  }
+
+  librbd::Image image;
+  r = utils::open_image_by_id(dst_io_ctx, migration_status.dest_image_id,
+                              true, &image);
+  if (r < 0) {
+    return r;
+  }
+
+  std::vector<librbd::snap_info_t> snaps;
+  r = image.snap_list(snaps);
+  if (r < 0) {
+    std::cerr << "rbd: listing snapshots failed: " << cpp_strerror(r)
+              << std::endl;
+    return r;
+  }
+
+  for (auto &snap : snaps) {
+    r = image.snap_set_by_id(snap.id);
+    if (r < 0) {
+      std::cerr << "rbd: accessing destination snapshot " << snap.name << "("
+                << snap.id << ") failed: " << cpp_strerror(r) << std::endl;
+      return r;
+    }
+
+    std::vector<librbd::linked_image_spec_t> children;
```

Comment:
Refactor this into a recursive function -- or perhaps add a new API method to collect all decedent children for a given parent.

Reply:
@dillaman  I updated the `list_children` API function to work if the parent image snapshot is not specified, returning all decedent children in this case. Hope it is what you wanted.

Reply:
My concern is about all decedents from a parent image (i.e. parent -> child -> grandchild (-> great-grandchild). It might be nice to have a new ```list_descendants``` method like ```list_children``` that will also recursively open each child and look for it's children (if any).

Reply:
I see now. Just `list_children` would be enough to detect that some children exist, but `list_descendants` will give the user a full list images she needs to care about.

I will add and use `list_descendants`. What do you think about my current change to `list_children` API (and CLI)? Does it look useful? I could limit the API changes only to the newly added `list_children3`  if you thought we should not modify the behavior of the older API functions, which are deprecated anyway.